interfaces/kmod/backend.go

```diff
@@ -32,11 +53,65 @@ func (b *Backend) Name() string {
 	return "kmod"
 }
 
+// Setup creates a conf file with list of kernel modules required by given snap,
+// writes it in /etc/modules-load.d/ directory and immediately loads the modules
+// using /sbin/modprobe. The devMode is ignored.
+//
+// If the method fails it should be re-tried (with a sensible strategy) by the caller.
 func (b *Backend) Setup(snapInfo *snap.Info, devMode bool, repo *interfaces.Repository) error {
-	// TODO: get snippets, load modules, create /etc/modules-load.d/snap.modules.conf file
-	return nil
+	snapName := snapInfo.Name()
+	// Get the snippets that apply to this snap
+	snippets, err := repo.SecuritySnippetsForSnap(snapInfo.Name(), interfaces.SecurityKMod)
+	if err != nil {
+		return fmt.Errorf("cannot obtain kmod security snippets for snap %q: %s", snapName, err)
+	}
+	if len(snippets) == 0 {
+		// Make sure that the modules conf file gets removed when we don't have any content
+		return removeModulesFile(snapName)
+	}
+
+	modules := b.processSnipets(snapInfo, snippets)
+	err = writeModulesFile(modules, snapName)
+	if err == osutil.ErrSameState {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+	return loadModules(modules)
 }
 
+// Remove removes modules config file specific to a given snap.
+//
+// This method should be called after removing a snap.
+//
+// If the method fails it should be re-tried (with a sensible strategy) by the caller.
 func (b *Backend) Remove(snapName string) error {
+	removeModulesFile(snapName)
 	return nil
 }
+
+// processSnipets combines security snippets collected from all the interfaces
+// affecting a given snap into a de-duplicated list of kernel modules.
+func (b *Backend) processSnipets(snapInfo *snap.Info, snippets map[string][][]byte) (modules [][]byte) {
```

Comment:
You could change this function to a simple one like `uniqueLines([]string) []string` and decouple it from everything else. Might make it easier to test.